### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by nicholeous**
- Skeletons for urgently required pieces of API documentation
- Joining example docs
- alphabetized and added to summary
- API reference
- Document skeletons for any packages that have not been added to docs
- link to them in summary 
- correct location of interbit-middleware in summary

### CODE REVIEW

File: docs/gitbook/examples/joining.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+These actions can also be [redispatched](../reference/interbit-covenant-utils/redispatch.md) from within a smart contract.
+
+There are two ways to join chains together. Statically when the node is started from the configuration file and dynamically by dispatching an action during chain operation.
+
+When creating a join both sides must agree on the join name and both sides must dispatch a corresponding action.
+
+## By Configuration File
+
+Ultimately, the joins setup in the configuration file are dispatched by the admin keys to the chain during start and deployment.
```
Comment (karikushi) [inline docs/gitbook/examples/joining.md]: Replace 'setup' with 'set up' in first and third paragraph

File: docs/gitbook/reference/interbit-cli/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Interbit CLI
+
+Interbit provides scripts that work with an app that is either created with create-interbit-app (coming soon...) or installed to a project with npm. It aids in developing and deploying interbit applications.
```
Comment (karikushi) [inline docs/gitbook/reference/interbit-cli/README.md]: Replace interbit with Interbit

File: docs/gitbook/examples/joining.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      }
+    }
+  }
+  // ...
+}
+```
+
+## By Action Dispatch
+
+When dynamically dispatching join actions to a chain you must be sure to match the join names. Chain IDs must be used as the friendly alias is only available in the configuration file and the middleware.
```
Comment (karikushi) [inline docs/gitbook/examples/joining.md]: A comma after 'Chain IDs must be used' would be helpful

File: docs/gitbook/reference/interbit-cli/deploy.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#### Options
+
+1. `--manifest` *(Filepath)*: an Interbit manifest file used for deployment instructions. If none is provided it will be read from `{--artifacts}/interbit.manifest.json`
+1. `--artifacts` *(Filepath)*: a directory of build artifacts to deploy. Should contain a manifest file and packed covenants. If none is given the current working directory is used.
+1. `--port` *(number)*: The port number interbit will communicate on
+1. `--admin-keys` *(filepath)*: a js or json file exporting a key pair to be used to boot the hypervisor with. If none is given a random pair will be generated which may not be permissioned on the chain network.
+
+
+#### Example
+
```
Comment (djlmills) [inline docs/gitbook/reference/interbit-cli/deploy.md]: Strictly speaking this isn't an example as you still have placeholders.  Make up the values in [] or change the label to usage and be consistent with --port [port]

File: docs/gitbook/reference/interbit-covenant-utils/destroy.md [hunk 5]
```diff
@@ -17,7 +17,13 @@ An action that can be dispatched to a chain to destroy it.
 #### Example
```
Comment (djlmills) [inline docs/gitbook/reference/interbit-covenant-utils/destroy.md]: Why are we documenting `interbit-covenant-utils`? This functionality is exposed in `interbit-covenant-tools` and you have used tools in all your other examples. We have done everything we can to keep users from needing to work with `interbit-core` and `interbit-covenant-utils` directly and there are only 4 direct references in `package.json`.

Comment (brentbenton) [reply docs/gitbook/reference/interbit-covenant-utils/destroy.md]: Fair, but can be addressed subsequently.

**Pull request merged**